flash/vision/style_transfer/model.py

```
@@ -0,0 +1,185 @@
+
+        return output
+
+
+class Residual(nn.Module):
+
+    def __init__(self, channels: int) -> None:
+        super().__init__()
+        self.conv1 = Conv(channels, channels, kernel_size=3)
+        self.conv2 = Conv(channels, channels, kernel_size=3, activation=False)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        output = self.conv2(self.conv1(input))
+        return output + input
+
+
+class Transformer(nn.Module):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.encoder = nn.Sequential(
+            Conv(3, 32, kernel_size=9),
+            Conv(32, 64, kernel_size=3, stride=2),
+            Conv(64, 128, kernel_size=3, stride=2),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+        )
+        self.decoder = nn.Sequential(
+            Conv(128, 64, kernel_size=3, stride=2, upsample=True),
+            Conv(64, 32, kernel_size=3, stride=2, upsample=True),
+            Conv(32, 3, kernel_size=9, norm=False, activation=False),
+        )
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        return self.decoder(self.encoder(input))
+
+
+class StyleTransfer(Task):
+
+    def __init__(
+        self,
+        style_image: torch.Tensor,
+        model: Optional[nn.Module] = None,
+        multi_layer_encoder: Optional = None,
+        content_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        style_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        optimizer: Union[Type[torch.optim.Optimizer], torch.optim.Optimizer] = torch.optim.Adam,
+        optimizer_kwargs: Optional[Dict[str, Any]] = None,
+        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
+        scheduler_kwargs: Optional[Dict[str, Any]] = None,
+        metrics: Union[torchmetrics.Metric, Mapping, Sequence, None] = None,
```

I see no point in accepting this, since there are no metrics for style transfer.

---

I would keep it for consistency with others task and maybe raise an error with a meaningful message. We don't know in future if anyone will come with goof metric for this task

---

Besides, what do you look at when you train style transfer - just the loss ?

---

Yeah, I would remove it then just add it back if one day there is a metric